text/0000-api-doc-frontpage-styleguide.md

```diff
@@ -0,0 +1,174 @@
+a good place to introduce the what and the why the crate. A good introduction is concise but also
+provides enough information for the reader to understand the purpose of the crate. In addition, a
+good introduction provides a learning path for the user as they read the rest of the documentation.
+
+Here's an example of a good introduction from the [log
+crate](https://doc.rust-lang.org/log/log/index.html):
+    
+> A lightweight logging facade.  
+>
+> A logging facade provides a single logging API that abstracts over the actual logging
+> implementation. Libraries can use the logging API provided by this crate, and the consumer
+> of those libraries can choose the logging framework that is most suitable for its use case.
+>
+> If no logging implementation is selected, the facade falls back to a "noop" implementation
+> that ignores all log messages. The overhead in this case is very small - just an integer load,
+> comparison and jump.  > > A log request consists of a target, a level, and a body. A target
+> is a string which defaults to the module path of the location of the log request, though that
+> default may be overridden. Logger implementations typically use the target to filter requests
+> based on some user configuration.
+
+There are several things that make this a good introduction section:
+    
+* A concise explanation of the crate's purpose.
+* The summary describes how the crate behaves in 3 short paragraphs.
+* Even if you don't want to read the rest of the `README`, you have enough to get started.
+* Readable by a layman, no jargon is introduced in the first few paragraphs.
+    
+What do we mean by "no jargon is introduced in the first few paragraphs"? Using the `rand` crate as
+an example - the crate's initial documentation probably should _not_ include a discussion of
+cryptographically secure random number generators since its primary purpose, for most Rust
+developers, will be to produce any random number. This information should instead be provided in a
+separate section that specifically discusses cryptographically secure random number generation.
+
+In general - it's good to give a developer a place to easily learn what your crate is about. Once
+they have the general idea, you can dive into more details that are specific to your crate,
+including jargon that would be common with its use.
+
+## First Example
+
+The first example should provide a concise example to help a developer understand how they can work
+with your crate. Ideally, this example should only demonstrate your crate and core Rust
+functionality. Avoid large examples with lots of features; larger examples are better fit for the
+`examples` directory.
+
+Let's look at a sample of a good first example of a crate. Notice that the author has focused on
+getting started, showing how to import and use the crate, and a few simple uses of common
+functionality.
```

Comment:
Lately I’ve begun to also include a snippet that may be copy-pasted into a `Cargo.toml` (reasons in next paragraph). I’d suggest including that into this recommendation as well.

I do so, because the first thing I check out about the crate is its documentation. Seeing something like
``````
//! ```toml
//! [build-dependencies]
//! gcc = "0.3"
//! ```
```````

saves me the effort of:

* remembering the crate name (e.g. crate is gcc, repository is gcc-rs; causes confusion);
* figuring out what the latest version is;
* what section I am supposed to put the dependency into;

The last point is esp. important when adding a dependency to a project which had no dependencies before, or adding the dependency in non-trivial ways (say with some features enabled/disabled) – I have trouble remembering how to do those things all the time.

---

Basically my first example becomes a `Usage` instead with step-by-step instructions on how to use a crate.